### ISSUE TITLE
508-defect-1 ❗ Launchblocker [DATA SAFETY]: If an email address is provided, it MUST be validated

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12585]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

For the email address field, the re-enter field is intended to confirm the correct email address is entered. If an email is entered in the first field, the user **must** re-enter the email the second field. The re-enter field should become a required field once the first field contains an email address, and the user should be unable to continue. The entry in the first email address field should match the entry in the second field.

Due to the potential harm of accidentally sending personal information to a stranger, this is considered a 508-defect-1 launchblocker issue.

This applies to each individual that provides an email address (the Veteran or service member, and each Caregiver).

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a 10-10CG online applicant, I want to provide an email address and be protected from mistakenly entering a wrong address.

## Environment

* Operating System: all
* Browser: all
* Server destination: production

## Steps to Recreate

1. Enter https://staging.va.gov/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/ in browser
2. Navigate to the **Veteran or service member information** section
3. Enter an email address in the first email address input, but leave the second one blank
4. Select Continue
5. Verify that it is possible to continue the form without validating the email address match in the first and second inputs

## Possible Fixes (optional)

- Recommend creating logic that the two email address input fields must match before a user may proceed.
- Consider pattern matching that the email entered is a well-formed email
- Provide useful, plain language error messages
    * If the second field is empty
    * If the email address is not a well-formed email

## WCAG or Vendor Guidance (optional)

Note: WUHCAG links provided additionally because they are generally more plain language, easier to understand.

* [W3C WCAG - Success Criterion 3.3.4 Error Prevention (Legal, Financial, Data) (Level AA):](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html)

* [W3C WCAG - Error Prevention (Legal, Financial, Data): Understanding SC 3.3.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-reversible.html)

* [WUHCAG - 3.3.4 – Error Prevention (Legal, Financial, Data) (Level AA)](https://www.wuhcag.com/error-prevention-legal-financial-data/)

* [W3C WCAG - 3.3.6 Error Prevention (All) (Level AAA)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-reversible-all.html)

* [WUHCAG - 3.3.6 – Error Prevention (All) (Level AAA)](https://www.wuhcag.com/error-prevention-all/)

## Screenshots or Trace Logs

![Screen Shot 2020-08-18 at 11 40 28 AM](https://user-images.githubusercontent.com/57469/90534634-f8201080-e147-11ea-988a-fb3bf3b93a3a.png)

**Comment by jenstrickland** [comment]
Validated. 🎉

**Comment by jenstrickland** [comment]
While auditing with JAWS and IE11, I noticed this behavior with the "at least 6 characters". Is this part of standard email validation on VA.gov? If so, I'll re-close the ticket. It was confusing, but if this is how we validate emails then it's manageable.

![image (7)](https://user-images.githubusercontent.com/57469/91331934-75c3cc00-e799-11ea-8e50-860514520545.png)

![image (8)](https://user-images.githubusercontent.com/57469/91331953-7bb9ad00-e799-11ea-91d1-b08e530a317b.png)

**Comment by LinnJS** [comment]
@jenstrickland ready for validation

**Comment by jenstrickland** [comment]
Validated 🎉
